src/precice/impl/SolverInterfaceImpl.cpp

```diff
@@ -934,6 +967,8 @@ void SolverInterfaceImpl::writeBlockVectorData(
     const double *values)
 {
   PRECICE_TRACE(fromDataID, size);
+  PRECICE_CHECK(_state != State::Constructed, "writeBlockVectorData(...) has to be called after initialize().");
```

Comment:
```suggestion
  PRECICE_CHECK(_state != State::Constructed, "writeBlockVectorData(...) can only be called after initialize().");
```
Otherwise it sounds like the function should only be called once. And that is has to be called.
Same for other examples below.